### PR TITLE
Refactor/30442/varification step

### PR DESCRIPTION
**Pull request by barttom**
### Fixed Issues

$ https://github.com/Expensify/App/issues/30442
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
        - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x]  If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added Design label so the design team can review the changes.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/7682108/a08640d0-70ce-40ca-9236-da4d4297dfe2
32666_android_s2
32666_android_s1

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

32666_ios_s1
32666_ios_s2

https://github.com/Expensify/App/assets/7682108/98fcbea4-2006-43cd-852d-c405477c57ba

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/7682108/0f505d00-0080-4cfb-9c96-60a0717d5bab

32666_ios_web_s1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/7682108/ba865806-9bf8-482c-810b-fda789c24631
32666_web_s2
32666_web_s1

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/7682108/e798b810-8012-449f-a1fd-eb7639be0fe6

32666_desktop_s1
32666_desktop_s2

</details>

### CODE REVIEW

**Comment by gedu** [comment]
This PR depends on https://github.com/Expensify/App/pull/32709

File: src/pages/ReimbursementAccount/CompleteVerification/substeps/ConfirmAgreements.js [hunk 1]
```diff
@@ -21,16 +37,34 @@ const validate = (values) => {
...
-        errors.isControllingOfficer = 'completeVerificationStep.isControllingOfficerError';
+    if (!ValidationUtils.isRequiredFulfilled(values.isAuthorizedToUseBankAccount)) {
+        errors.isAuthorizedToUseBankAccount = 'completeVerificationStep.isAuthorizedToUseBankAccountError';
     }
 
     return errors;
 };
 
-function ConfirmAgreements() {
+const completeVerificationKeys = CONST.BANK_ACCOUNT.COMPLETE_VERIFICATION.INPUT_KEY;
```
Comment (koko57) [inline src/pages/ReimbursementAccount/CompleteVerification/substeps/ConfirmAgreements.js:47]: NAB: in the previous file you save the same CONST.BANK_ACCOUNT.COMPLETE_VERIFICATION.INPUT_KEY as COMPLETE_VERIFICATION_KEYS  (in uppercase) and here in camelCase, maybe let's stick to the one convention?

File: src/pages/ReimbursementAccount/ConnectBankAccount/components/Enable2FACard.js [hunk 2]
```diff
@@ -1,50 +1,47 @@
...
-
-    /** policyID of the workspace where user is setting up bank account */
-    policyID: PropTypes.string.isRequired,
-};
-
-function Enable2FAPrompt({translate, policyID}) {
+function Enable2FACard() {
     const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const secureYourAccountUrl = encodeURI(
```
Comment (koko57) [inline src/pages/ReimbursementAccount/ConnectBankAccount/components/Enable2FACard.js:15]: can we have it outside the component, or not really?

File: src/pages/ReimbursementAccount/ACHContractStep.js [hunk 3]
```diff
@@ -13,9 +13,12 @@ const propTypes = {
 
 function ACHContractStep({onBackButtonPress, onCloseButtonPress}) {
     const [isBeneficialOwnerInfoSet, setIsBeneficialOwnerInfoSet] = useState(false);
+    const handleCompleteVerificationBackButtonPress = useCallback(() => {
+        setIsBeneficialOwnerInfoSet(false);
+    }, []);
 
     if (isBeneficialOwnerInfoSet) {
-        return <CompleteVerification setIsBeneficialOwnerInfoSet={setIsBeneficialOwnerInfoSet} />;
+        return <CompleteVerification onBackButtonPress={handleCompleteVerificationBackButtonPress} />;
```
Comment (MrMuzyk) [inline src/pages/ReimbursementAccount/ACHContractStep.js:21]: onBackButton press should come from above via props. It got updated in all the other steps somewhere along the way. Also you need to pass handler and display `X` button on the header inside CompleteVerification component. It should be similar to BeneficialOwnerInfo

File: src/pages/ReimbursementAccount/CompleteVerification/substeps/ConfirmAgreements.js [hunk 4]
```diff
@@ -21,16 +37,38 @@ const validate = (values) => {
...
     return errors;
 };
 
-function ConfirmAgreements() {
+const COMPLETE_VERIFICATION_KEYS = CONST.BANK_ACCOUNT.COMPLETE_VERIFICATION.INPUT_KEY;
+
+function ConfirmAgreements({onNext, reimbursementAccount}) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
+    const defaultValues = useMemo(
```
Comment (MrMuzyk) [inline src/pages/ReimbursementAccount/CompleteVerification/substeps/ConfirmAgreements.js:52]: IMO memo is an overkill here. Its just lodashGet inside of that getDefaultValueForReimbursementAccountField method

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    account: {
+        requiresTwoFactorAuth: false,
+    },
+};
+
+function ConnectBankAccount({reimbursementAccount, onBackButtonPress, account}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const bankAccountState = lodashGet(reimbursementAccount, 'achData.state');
```
Comment (MrMuzyk) [inline src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:43]: no default values for all the gets?

File: src/pages/ReimbursementAccount/EnableBankAccount/EnableBankAccount.js [hunk 6]
```diff
@@ -32,46 +31,43 @@ const propTypes = {
...
 };
 
-function EnableStep(props) {
+function EnableBankAccount({reimbursementAccount, user, onBackButtonPress}) {
     const styles = useThemeStyles();
-    const isUsingExpensifyCard = props.user.isUsingExpensifyCard;
-    const achData = lodashGet(props.reimbursementAccount, 'achData') || {};
+    const {translate} = useLocalize();
+
+    const achData = lodashGet(reimbursementAccount, 'achData') || {};
```
Comment (MrMuzyk) [inline src/pages/ReimbursementAccount/EnableBankAccount/EnableBankAccount.js:46]: move the `{}` inside of lodashGet as default returned value

File: src/pages/ReimbursementAccount/EnableBankAccount/EnableBankAccount.js [hunk 7]
```diff
@@ -32,46 +31,43 @@ const propTypes = {
...
     const {icon, iconSize} = getBankIcon({bankName: achData.bankName, styles});
-    const formattedBankAccountNumber = achData.accountNumber ? `${props.translate('paymentMethodList.accountLastFour')} ${achData.accountNumber.slice(-4)}` : '';
+    const isUsingExpensifyCard = user.isUsingExpensifyCard;
+    const formattedBankAccountNumber = achData.accountNumber ? `${translate('paymentMethodList.accountLastFour')} ${achData.accountNumber.slice(-4)}` : '';
     const bankName = achData.addressName;
+    const errors = lodashGet(reimbursementAccount, 'errors', {});
+    const pendingAction = lodashGet(reimbursementAccount, 'pendingAction', null);
+    const shouldShowResetModal = lodashGet(reimbursementAccount, 'shouldShowResetModal');
```
Comment (MrMuzyk) [inline src/pages/ReimbursementAccount/EnableBankAccount/EnableBankAccount.js:53]: no default value

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js (hunk 5 above)
Comment (barttom) [reply src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:43]: it's copied from previous implementation, so not sure what default value supposed to be.

Comment (MrMuzyk) [reply src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:43]: surely any value in the matching type will be better than undefined?

Comment (barttom) [reply src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:43]: the question is what should be default 😄

Comment (MrMuzyk) [reply src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:43]: something that overlaps with the type of the value we're getting. So empty string, empty object or w/e

File: src/pages/ReimbursementAccount/ACHContractStep.js [hunk 8]
```diff
@@ -13,9 +13,17 @@ const propTypes = {
 
 function ACHContractStep({onBackButtonPress, onCloseButtonPress}) {
     const [isBeneficialOwnerInfoSet, setIsBeneficialOwnerInfoSet] = useState(false);
+    const handleCompleteVerificationBackButtonPress = useCallback(() => {
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ACHContractStep.js:16]: Do we really need useCallback here?

File: src/pages/ReimbursementAccount/CompleteVerification/CompleteVerification.js [hunk 9]
```diff
@@ -25,8 +25,11 @@ const propTypes = {
     /** The draft values of the bank account being setup */
     reimbursementAccountDraft: reimbursementAccountDraftPropTypes,
 
-    /** Changes variable responsible for displaying step 4 or 5 */
-    setIsBeneficialOwnerInfoSet: PropTypes.func.isRequired,
+    /** Handle back button press */
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/CompleteVerification/CompleteVerification.js:28]:
```suggestion
    /** Handles back button press */
```

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+import * as Report from '@userActions/Report';
+import ONYXKEYS from '@src/ONYXKEYS';
+import BankAccountValidationForm from './components/BankAccountValidationForm';
+import FinishChatCard from './components/FinishChatCard';
+
+const propTypes = {
+    /** Bank account currently in setup */
+    reimbursementAccount: ReimbursementAccountProps.reimbursementAccountPropTypes.isRequired,
+
+    onBackButtonPress: PropTypes.func.isRequired,
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:24]: Comment.

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    if (bankAccountState === BankAccount.STATE.OPEN) {
+        return (
+            <EnableBankAccount
+                reimbursementAccount={reimbursementAccount}
+                onBackButtonPress={onBackButtonPress}
+            />
+        );
+    }
+
+    const maxAttemptsReached = lodashGet(reimbursementAccount, 'maxAttemptsReached');
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:55]: Same - https://github.com/Expensify/App/pull/32666/files#r1429832194.

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    const isBankAccountVerifying = !maxAttemptsReached && bankAccountState === BankAccount.STATE.VERIFYING;
+    const isBankAccountPending = bankAccountState === BankAccount.STATE.PENDING;
+    const requiresTwoFactorAuth = lodashGet(account, 'requiresTwoFactorAuth');
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:58]: Same - https://github.com/Expensify/App/pull/32666/files#r1429832194.

File: src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+    return (
+        <ScreenWrapper
+            testID={ConnectBankAccount.displayName}
+            includeSafeAreaPaddingBottom={false}
+            shouldEnablePickerAvoiding={false}
+            shouldEnableMaxHeight
+            style={[styles.mh2]}
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ConnectBankAccount/ConnectBankAccount.js:66]:
```suggestion
            style={styles.mh2}
```

**Comment by barttom** [comment]
@shubham1206agra ready for review

**Comment by shubham1206agra** [comment]
Screenshot 2023-12-21 at 3 30 23 PM

The text is missing in color

**Comment by shubham1206agra** [comment]
Screenshot 2023-12-21 at 4 10 44 PM

Is this intentional? This happened when I clicked `Add work email address`

**Comment by shubham1206agra** [comment]
Screenshot 2023-12-21 at 4 29 25 PM

Screenshot 2023-12-21 at 4 30 02 PM

**Comment by barttom** [comment]
@shubham1206agra https://github.com/Expensify/App/pull/32666#issuecomment-1866032891

On main is the same, I didn't change the logic. I received this URL after clicking on the button
```
https://www.expensify.com.dev/settings?param=%7B%22section%22:%22account%22,%22openModal%22:%22secondaryLogin%22%7D&authToken=23C9B8CA1CAF9E18CD07A6E281F6D1F483135F019F8EE49AFF86B87113D42553A4FB36517D0E7F67756A377227915D9A897BB5C8BE7CFA377B8A1ED9CD35BA73CDB3022A6F59C292E8898E99C9ADEB0A4FA698336EF0C1AC18619578665C4BF01EB5FFAA99BF5B74A6492DE6A46E94D1E823254F94C04BD7A48E0FC2F6B352ADC23BB5BFEA37FC3FE27C723BC35476C505CCFD23734AA175A53473A379514404B4D9DAFC07D8EECF14FB278C4B782C330109B466E647020D485BA0F08C9CC3F84FB365828D1954AD82A902FC1C7E2D86E315E1D9431218768BABC36A31B1E915491973E7339D608B9795AE27DB55A640410C412CF6686A278F3DB7B4659C568C50E75FC5381862E06AC0945E7D5202200C9EE8A347EAF536A3233472B5400F32408BC3240FC128BFE0C3588E3F0B79E46B34CC27933ADB5EB7988C851EC6C4FD71BA918C7470FC9756856AB1D63A62AD6D8B5027A559FDD4CB47556BAD10CF0EFA333A954915F9DB6FF730AB40D65720&email=bartlomiej.tomczyk%2B30442%40callstack.com
```

**Comment by shubham1206agra** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/58412969/13c3a19b-93bb-44d4-b35e-5df339344baa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/58412969/d39df42c-a7d9-4c2b-b4b2-bd3c41ccaeec

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/58412969/44d69df1-2bb4-4fc9-91f5-112f2872bd04

Screenshot 2023-12-21 at 2 41 05 PM
Screenshot 2023-12-21 at 2 44 29 PM
Screenshot 2023-12-21 at 2 45 54 PM
Screenshot 2023-12-21 at 2 46 20 PM
Screenshot 2023-12-21 at 2 47 15 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/58412969/07be345d-3137-4b90-8b21-1035bc618f32

</details>

**Comment by shubham1206agra** [comment]
> @shubham1206agra [#32666 (comment)](https://github.com/Expensify/App/pull/32666#issuecomment-1866032891)
> 
> On main is the same, I didn't change the logic. I received this URL after clicking on the button
> 
> ```
> https://www.expensify.com.dev/settings?param=%7B%22section%22:%22account%22,%22openModal%22:%22secondaryLogin%22%7D&authToken=23C9B8CA1CAF9E18CD07A6E281F6D1F483135F019F8EE49AFF86B87113D42553A4FB36517D0E7F67756A377227915D9A897BB5C8BE7CFA377B8A1ED9CD35BA73CDB3022A6F59C292E8898E99C9ADEB0A4FA698336EF0C1AC18619578665C4BF01EB5FFAA99BF5B74A6492DE6A46E94D1E823254F94C04BD7A48E0FC2F6B352ADC23BB5BFEA37FC3FE27C723BC35476C505CCFD23734AA175A53473A379514404B4D9DAFC07D8EECF14FB278C4B782C330109B466E647020D485BA0F08C9CC3F84FB365828D1954AD82A902FC1C7E2D86E315E1D9431218768BABC36A31B1E915491973E7339D608B9795AE27DB55A640410C412CF6686A278F3DB7B4659C568C50E75FC5381862E06AC0945E7D5202200C9EE8A347EAF536A3233472B5400F32408BC3240FC128BFE0C3588E3F0B79E46B34CC27933ADB5EB7988C851EC6C4FD71BA918C7470FC9756856AB1D63A62AD6D8B5027A559FDD4CB47556BAD10CF0EFA333A954915F9DB6FF730AB40D65720&email=bartlomiej.tomczyk%2B30442%40callstack.com
> ```

@mountiny Shouldn't we use newdot contact page here?

**Comment by barttom** [comment]
https://github.com/Expensify/App/pull/32666#issuecomment-1866057337
This warning comes from Section component and It's about typing with prop-types. The thing is we need to put component into icon prop but is defined as a function. Don't think is a part of this refactor.

**Comment by barttom** [comment]
https://github.com/Expensify/App/pull/32666#issuecomment-1866057337
fixed

**Comment by shubham1206agra** [comment]
@barttom Can you fix the lint?

File: src/pages/ReimbursementAccount/BankInfo/substeps/Confirmation.js [hunk 14]
```diff
@@ -73,21 +74,23 @@ function Confirmation({reimbursementAccount, reimbursementAccountDraft, onNext})
...
                                 src={Expensicons.Bank}
-                                additionalStyles={[styles.confirmBankInfoCompanyIcon, styles.mb5]}
+                                additionalStyles={[styles.confirmBankInfoCompanyIcon, styles.mr3]}
                                 fill={theme.iconHovered}
                             />
-                            <View style={[styles.mb3]}>
-                                <Text style={[styles.mutedTextLabel, styles.mb1]}>{translate('bankAccount.routingNumber')}</Text>
-                                <Text style={styles.confirmBankInfoNumber}>{values[bankInfoStepKeys.ROUTING_NUMBER]}</Text>
-                            </View>
                             <View>
```
Comment (shubham1206agra) [inline src/pages/ReimbursementAccount/BankInfo/substeps/Confirmation.js:85]: Why do we have this `View` here?

**Review by shubham1206agra — APPROVED** [review]
LGTM!

**Comment by shubham1206agra** [comment]
Just get this merged after the lint fixes

**Comment by barttom** [comment]
@shubham1206agra lint fixed, and I answered for the question

File: src/pages/ReimbursementAccount/BankInfo/substeps/Confirmation.js (hunk 14 above)
Comment (shubham1206agra) [reply src/pages/ReimbursementAccount/BankInfo/substeps/Confirmation.js:84]: Thanks for the clarification

**Comment by shubham1206agra** [comment]
@barttom Add these points to checklist 
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.

**Comment by barttom** [comment]
@shubham1206agra Done, but I can't add labels to de PR  and changes in the form component I made were only fixes after broken merge main branch

**Comment by shubham1206agra** [comment]
We should take care of this in final PR

**Review by mountiny — APPROVED** [review]
Looks good, thanks!

**Pull request merged**